implementations/CMIP6-UofT/add_CMIP6.py

```diff
@@ -19,6 +22,116 @@
 LOGGER.propagate = False
 
 
+def collection2enum(collection):
+    """Create Enum based on terms from pyessv collection.
+
+    Parameters
+    ----------
+    collection : pyessv.model.collection.Collection
+      pyessv collection of terms.
+
+    Returns
+    -------
+    Enum
+      Enum storing terms and their labels from collection.
+    """
+    mp = {term.name: term.label for term in collection}
+    return Enum(collection.raw_name.capitalize(), mp, module="base")
+
+
+# CMIP6 controlled vocabulary (CV)
+CV = pyessv.WCRP.CMIP6
+
+# Enum classes built from the pyessv' CV
+Activity = collection2enum(CV.activity_id)
+Experiment = collection2enum(CV.experiment_id)
+Frequency = collection2enum(CV.frequency)
+GridLabel = collection2enum(CV.grid_label)
+Institute = collection2enum(CV.institution_id)
+Member = collection2enum(CV.member_id)
+Resolution = collection2enum(CV.nominal_resolution)
+Realm = collection2enum(CV.realm)
+Source = collection2enum(CV.source_id)
+SourceType = collection2enum(CV.source_type)
+SubExperiment = collection2enum(CV.sub_experiment_id)
+Table = collection2enum(CV.table_id)
+Variable = collection2enum(CV.variable_id)
+
+
+class Properties(BaseModel):
+    """Data model for CMIP6 Controlled Vocabulary.
+    """
+    activity: Activity = Field(..., alias="activity_id")
+    experiment: Experiment = Field(..., alias="experiment_id")
+    frequency: Frequency
+    grid_label: GridLabel
+    institute: Institute = Field(..., alias="institute_id")
+    member: Member = Field(..., alias="member_id")
+    resolution: Resolution = Field(..., alias="nominal_resolution")
+    realm: Realm = Field(..., alias="realm")
+    source: Source = Field(..., alias="source_id")
+    source_type: SourceType = Field(..., alias="source_type")
+    sub_experiment: SubExperiment = Field(..., alias="sub_experiment_id")
+    table: Table = Field(..., alias="table_id")
+    variable: Variable = Field(..., alias="variable_id")
+    initialization_index: int
+    physics_index: int
+    realization_index: int
+    forcing_index: int
+    variant_label: str
+    version: str
+    license: str = None
+    grid: str = None
+    tracking_id: str = Field(..., alias="tracking_id")
+
+
+def ncattrs_to_geometry(attrs: MutableMapping[str, Any]) -> MutableMapping[str, Any]:
```

Comment:
I wonder if we should make this a class method for `THREDDSLoader` so that they can be reused. The input keys these are expecting are very tied to the data returned by the THREDDS server.

Reply:
Makes sense, will do. Didn't want to integrate things to tightly until we got something working.